src/Apps/Order/Routes/Shipping2/ShippingContext.tsx

```diff
@@ -0,0 +1,176 @@
+  newSavedAddressId: string | null
+  selectedShippingQuoteId: string | null
+  stage: ShippingStage
+  isPerformingOperation: boolean
+  isArtsyShipping: boolean
+}
+
+interface Actions {
+  dialog: Dialog
+  handleExchangeError: (
+    error: {
+      code: string
+      data: string | null | undefined
+    },
+    logger: ReturnType<typeof createLogger>
+  ) => void
+  setFormHelpers: (payload: FormHelpers) => void
+  setSelectedShippingQuote: (payload: string | null) => void
+  setNewSavedAddressId: (payload: string | null) => void
+  setIsPerformingOperation: (payload: boolean) => void
+  setStage: (payload: ShippingStage) => void
+}
+
+export interface ShippingContextProps {
+  actions: Actions
+  state: State
+  orderData: ComputedOrderData
+}
+
+export const ShippingContext = createContext<ShippingContextProps>({} as any)
+
+export const ShippingContextProvider: FC<Pick<
+  ShippingProps,
+  "order" | "me" | "dialog"
+>> = props => {
+  const orderData = computeOrderData(props.order, props.me)
+
+  const isArtsyShipping = !!orderData.savedFulfillmentDetails?.isArtsyShipping
+
+  const initialState: State = {
+    /**
+     * Because there is a single button for both fulfillment details and
+     * shipping quote steps (and duplicated in the sidebar)
+     * we need to hack some formik values UP from the fulfillment details form.
+     */
+    formHelpers: {
+      // Used to submit the form
+      submitForm: () => Promise.reject(new Error("form not loaded")),
+      // Used to disable the button
+      isValid: false,
+      // Used to get the form values for un-saving the address if the user
+      // unchecks it after saving it in the fulfillment details step.
+      values: ({
+        attributes: {
+          saveAddress: false,
+        },
+      } as unknown) as FulfillmentValues,
+    },
+    isArtsyShipping,
+    isPerformingOperation: false,
+    newSavedAddressId: null,
+    selectedShippingQuoteId: orderData.selectedShippingQuoteId ?? null,
+    stage: isArtsyShipping ? "refresh_shipping_quotes" : "fulfillment_details",
+  }
+
+  const [state, dispatch] = useReducer(shippingStateReducer, initialState)
+
+  const { handleExchangeError } = useHandleExchangeError({
+    dialog: props.dialog,
+    isArtsyShipping,
+    orderData,
+  })
+
+  const actions = {
+    setFormHelpers: (formHelpers: FormHelpers) => {
+      dispatch({ type: "SET_FORM_HELPERS", payload: formHelpers })
+    },
+    setSelectedShippingQuote: (payload: string | null) => {
+      dispatch({ type: "SET_SELECTED_SHIPPING_QUOTE", payload })
+    },
+    setNewSavedAddressId: (payload: string | null) => {
+      dispatch({ type: "SET_NEW_SAVED_ADDRESS_ID", payload })
+    },
+    setStage: (payload: ShippingStage) => {
+      dispatch({ type: "SET_STAGE", payload })
+    },
+    setIsPerformingOperation: (payload: boolean) => {
+      dispatch({ type: "SET_IS_PERFORMING_OPERATION", payload })
+    },
+    handleExchangeError,
+    dialog: props.dialog,
```

Comment:
Not a huge deal, but dialog is actually an object more than an action

Reply:
Thought about this but settled on it being ok because inside of `dialog` there are only action-like function calls that do things, so this just namespaces those